src/state/usePasscodeAuth/usePasscodeAuth.ts

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+export function getPasscode() {
+  const match = window.location.search.match(/passcode=(.*)&?/);
+  const passcode = match ? match[1] : window.sessionStorage.getItem('passcode');
+  return passcode;
+}
+
+export function fetchToken(name: string, room: string, passcode: string) {
+  return fetch(`/token`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({ user_identity: name, room_name: room, passcode }),
+  });
+}
+
+export function verifyPasscode(passcode: string) {
+  return fetchToken('verification name', 'verification room', passcode).then(async res => {
+    const jsonResponse = await res.json();
+    if (res.status === 401) {
+      return { isValid: false, error: jsonResponse.error?.message };
+    }
+
+    if (res.ok && jsonResponse.token) {
+      return { isValid: true };
+    }
+  });
+}
+
+export function getErrorMessage(message: string) {
+  switch (message) {
+    case 'passcode incorrect':
+      return 'Passcode is incorrect';
+    case 'passcode expired':
+      return 'Passcode has expired';
+    default:
+      return message;
+  }
```

Comment:
This function seems weird. Should we define error codes instead? This comes from the twilio function endpoint right?

Reply:
The `message` property on the `error` object is basically serving as an error code. We decided on this format here: https://github.com/twilio-labs/plugin-rtc/pull/3#discussion_r387772234

Reply:
Ok, I left a comment in that PR.